### ISSUE TITLE
[Feature Request] Add support for digital music files

### ISSUE THREAD

**Issue by starslab** [1oom-fork/1oom#12]
I would love to see optional support for digital music packs for 1oom, similar to how OpenXCom implements such things.

That would allow, among others:
* me to publish a completely authentic Roland LA32 music pack - see https://www.youtube.com/playlist?list=PLQfcsYP5T4-Ezqrluq_F-oCa0epgYAyxI
* Someone else to perhaps publish a highly-authentic OPL music pack that some people would probably like to have

Instead of mucking about with MIDI, just start playing the appropriate digital audio file when requested by the game. Volume may be managed exactly as the original game does. If you implement LOOPSTART and/or LOOPEND metadata support from SDL_Mixer2, that would just make it even better.

**Comment by 1oom-fork** [comment]
Is it possible to just play the game in DOSBOX with Munt support?

**Comment by starslab** [comment]
Sure. But then you lose all of 1oom quality-of-life improvements.

**Comment by 1oom-fork** [comment]
I mean 1oom + dosbox

**Comment by starslab** [comment]
Right. sorry.

This idea of playing a "modern" game like 1oom in an actual DOS environment is so foreign to me it didn't even process.

Unless 1oom actually implements all the Roland logic (which I very highly doubt it does) that's not going to work. The MT-32 is *not* a General MIDI device.

**Comment by 1oom-fork** [comment]
I have no knowledge in this area, so it will take me some time to estimate the amount of work. I'd rather have a specialist do this as I have a lot of higher priority tasks. Feel free to make an effort if you wish.

**Comment by starslab** [comment]
It looks like some groundwork has already been done?

from `src/hw/sdl/hwdsl_audio.c`
```
static Mix_MusicType mus_type_to_sdlm(mus_type_t type)
{
    switch (type) {
        case MUS_TYPE_LBXXMID:
        case MUS_TYPE_MIDI:
            return MUS_MID;
        case MUS_TYPE_WAV:
            return MUS_WAV;
        case MUS_TYPE_OGG:
            return MUS_OGG;
        case MUS_TYPE_FLAC:
            return MUS_FLAC;
        default:
            return MUS_NONE;
    }
}
```

**Comment by starslab** [comment]
!!!

`doc/pbxin_example.txt`
```
# Replace the battle music with something fitting.
2,music.lbx,8,D_RUNNIN.mid
# Replace the "click" sound.
2,soundfx.lbx,0x24,ping.wav
# Tip: hex numbers are supported. Unfortunately, so are octal; 012 == 10 == 0xa.
# Currently supported music formats: MIDI, XMID (used in MOO1 and MOM), FLAC, Ogg Vorbis
# Currently supported sound formats: WAV, VOC

```

**Comment by starslab** [comment]
It verks!

It exposes a hopefully-easy to fix bug, in that looping music (At least, the main menu music) does not loop when replaced by an Ogg file.

The original XMIDI data in MUSIC.LBX contains MIDI looping instructions, so I'm guessing the game itself never actually had any logic devoted to looping music.

**Comment by 1oom-fork** [comment]
Thanks, that's much easier

**Comment by starslab** [comment]
Yeah, I'm going to strongly suggest you implement Ogg LOOPSTART meta-data support, as found in SDL-Mixer2 and RPGMaker - not only will my music packs sound slightly better than without it, but that actually keeps to the original design and game logic of 1oom.

This works, but is kinda crashy for me. I have no idea if it's crashing because of the music packs, or because of something else. I'm going to publicly post these and see if I can get feedback from others on that count.

Thanks for putting up with me!

**Comment by 1oom-fork** [comment]
The PBX function must be checked/tested for vulnerabilities i guess. Unfortunately i can't say it's 100% safe.
